packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -519,9 +519,27 @@ export class JobDefinition extends Resource implements IJobDefinition {
       return undefined;
     }
 
+    // If the AWS_*** environment variables are not explicitly set to the container, infer them from the current environment.
+    // This makes the usage of tools like AWS SDK inside the container frictionless
+
+    const environment = this.deserializeEnvVariables(container.environment);
+
+    if (!environment.find((x) => x.name === 'AWS_REGION')) {
+      environment.push({
+        name: 'AWS_REGION',
+        value: Stack.of(this).region,
+      });
+    }
+    if (!environment.find((x) => x.name === 'AWS_ACCOUNT')) {
+      environment.push({
+        name: 'AWS_ACCOUNT',
+        value: Stack.of(this).account,
+      });
+    }
+
     return {
       command: container.command,
-      environment: this.deserializeEnvVariables(container.environment),
+      environment,
```

Comment:
So, I'm wondering if always having values here is a breaking change for customers currently using this. Breaking changes are OK in an experimental module, but I'm wondering if we should make setting these defaults an option in the props instead of just doing it 100% of the time. I'm also wondering if there are use cases where a user would want this undefined. I'm not super familiar with AWS Batch so I think I need to consult with others on this.

Reply:
I'm not 100% sure about this specific scenario, but generally speaking, from my experience - when we talk about breaking change in code - it's usually a removal of parameters, functions, classes or a modification of their existing behavior, whereas additions of fields, parameters, methods are not considered a breaking change.

But again, that' just my perspective, looking at other libraries I've consumed in my projects as a developer. With CDK it's slightly different since we are considering not just the code but also things that users might have deployed using that code, and I realize it's a slightly more flavored and opiniated discussion.